### ISSUE TITLE
Parsing errors for snowflake syntax

### ISSUE THREAD

**Issue by ahladkyi-spd** [sqlfluff/sqlfluff#2206]
Getting parsing errors on snowflake queries

```
SELECT TOP 1 * FROM MYTABLE;

ALTER TABLE MYTABLE RENAME COLUMN MYCOLUMN1 TO MYCOLUMN2;

```

## Expected Behaviour
No errors, since it is a valid syntax

## Observed Behaviour
```
sqlfluff lint test.sql
== [test.sql] FAIL
L:   1 | P:  13 |  PRS | Line 1, Position 13: Found unparsable section: ' *'
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'ALTER
                       | TABLE MYTABLE RENAME COLUMN MYCOLU...'
```

## Steps to Reproduce
`sqlfluff lint --dialect snowflake --ignore linting`
## Dialect
snowflake

## Version
sqlfluff version
0.9.0

## Configuration
default

**Comment by wong-codaio** [comment]
@jpy-git @tunetheweb - Have been thinking about how to fix the `ALTER TABLE MYTABLE RENAME COLUMN MYCOLUMN1 TO MYCOLUMN2;` reported here - Wanted to get you guys' thoughts on the approach before I go in.

tl;dr Start implement a Snowflake-specific AlterTableStatement, mirror after grammar defined in [Snowflake's doc](https://docs.snowflake.com/en/sql-reference/sql/alter-table.html)

**Context**
1. Snowflake dialect currently reuses Ansi's [AlterTableStatement](https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_ansi.py#L2472) for most of its ALTER TABLE statement (The only exception is [AlterTableColumn](https://github.com/sqlfluff/sqlfluff/blob/043011fd9dd3d31d504800c2c8cfa40cd04154b1/src/sqlfluff/dialects/dialect_snowflake.py#L896)].
2. Reading [Snowflake's docs](https://docs.snowflake.com/en/sql-reference/sql/alter-table.html), there're quite a bit to the `ALTER TABLE` statement than what Ansi supports, and Snowflake definitely defines the grammar well in this case that makes sense to split it even further (Example: `clusteringAction` vs `tableColumnAction` vs `constraintAction`.)
3. It feels right to start implementing a Snowflake-specific `ALTER TABLE` statement.

**Plan**
1. To start, it'll be a simple copy-and-pasted version from Ansi, combined with with the Snowflake-specific `AlterTableColumn`. (No functionality change - Just to get the skeleton ready.)
2. Implement _some of_ `tableColumnAction` in the grammar implemented, so we can address _this_ specific bug.
3. Get `tableColumnAction` fully implemented (There're quite a few items.)
4. We can opportunistically pick up other pieces of grammar to be implemented as needed.

Thoughts?

**Comment by tunetheweb** [comment]
Sounds good to me. Only ask is that if only implementing some of the grammar then add a TODO comment to explain that.

**Comment by jpy-git** [comment]
@wong-codaio a very sensible plan 👍

**Comment by wong-codaio** [comment]
@jpy-git @tunetheweb - FYI: Part 1: https://github.com/sqlfluff/sqlfluff/pull/2267

**Comment by jpy-git** [comment]
@wong-codaio FYI this was closed by #2327 since it addressed the bug in this issue, but feel free to reopen or copy step 3 of your `ALTER TABLE` plan to a separate issue if you want :)
